File: PaymentProcessor/PayPalPaymentProcessor.ts
import PaymentProcessor from "./PaymentProcessor";

export default class PayPalPaymentProcessor extends PaymentProcessor {
  validate(): void {
    console.log("Validating paypal address");
  }
  performPayment(amount: number, currency: string): void {
    console.log("Making paypal payment");
  }
}

File: labs/lab02/lab2/templatemethodlab2/src/main/java/with/templatemethod/PaypalPaymentProcessor.java
package with.templatemethod;

import without.templatemethod.Customer;

public class PaypalPaymentProcessor extends PaymentProcessor {

    public PaypalPaymentProcessor(Customer customer) {
        super(customer);
    }

    @Override
    protected void validate() {
        System.out.println("Validating PayPal account details...");
    }

    @Override
    protected void performPayment(double amount, String currency) {
        System.out.println("Processing PayPal payment of " + amount + " " + currency);
    }

    @Override
    protected void sendReceipt() {
        System.out.println("Sending PayPal payment receipt to " + customer.getName() + " at " + customer.getEmail());
    }
}
